### PR TITLE
[CIAPP] Agent configuration preset when CI Visibility is enabled

### PR DESCRIPTION
**Pull request by drodriguezhdez**
This PR adds the logic to configure the `Agent` disabling all the agent features but CI Visibility if this one is enabled.
The override only happens if the user does not explicitly set the property.

Additionally, the `dd.prioritization.type` is set to `ENSURE_TRACE` if CI Visibility is enabled.

### CODE REVIEW

File: dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/Agent.java [hunk 1]
```diff
@@ -78,14 +104,29 @@
...
+      // if CI Visibility is enabled, all the other features are disabled by default
+      // unless the user had explicitly enabled them.
+      setSystemPropertyDefault(AgentFeature.TRACING.getSystemProp(), "false");
+      setSystemPropertyDefault(AgentFeature.JMXFETCH.getSystemProp(), "false");
+      setSystemPropertyDefault(AgentFeature.PROFILING.getSystemProp(), "false");
+      setSystemPropertyDefault(AgentFeature.APPSEC.getSystemProp(), "false");
+      setSystemPropertyDefault(AgentFeature.CWS.getSystemProp(), "false");
+
+      /*if CI Visibility is enabled, the PrioritizationType should be {@code Prioritization.ENSURE_TRACE} */
+      setSystemPropertyDefault("dd.prioritization.type", "ENSURE_TRACE");
```
Comment (drodriguezhdez) [inline dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/Agent.java:123]: Not sure if this is the best place to do this.

File: dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/Agent.java [hunk 2]
```diff
@@ -717,63 +758,21 @@ private static boolean isDebugMode() {
...
-   *
-   * @return true if we should
-   */
-  private static boolean isStartupLogsEnabled() {
-    final String startupLogsSysprop = "dd.trace.startup.logs";
-    String startupLogsEnabled = System.getProperty(startupLogsSysprop);
-    if (startupLogsEnabled == null) {
-      startupLogsEnabled = ddGetEnv(startupLogsSysprop);
+  /** @return {@code true} if the agent feature is enabled */
+  private static boolean isFeatureEnabled(AgentFeature feature) {
```
Comment (drodriguezhdez) [inline dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/Agent.java:762]: I refactored all the `is****Enabled` method to this. Let me know what you think.

**Comment by devinsba** [comment]
I think you want to set the base branch to the project branch

**Comment by drodriguezhdez** [comment]
> I think you want to set the base branch to the project branch

Yeah, I'm waiting for CI to finish to rebase `master` into the `project/ci-library` branch.

**Pull request closed without merging**